R/zchunk_LB123.Electricity_china.R

```diff
@@ -0,0 +1,293 @@
+             wind = wind + wind.adj,
+             nuclear = nuclear + nuc.adj,
+             solar = solar + solar.adj) %>%
+      subset(select = c(-coal.adj, -wind.adj, -nuc.adj, -solar.adj)) %>%
+      mutate(biomass = coal) %>%
+      gather(fuel, value, c(-province, -year)) %>%
+      filter(value == 0 | value > 0.1) %>%
+      complete(nesting(province, fuel), year = HISTORICAL_YEARS_c) %>%
+      arrange(province, fuel, year) %>%
+      group_by(province, fuel) %>%
+      mutate(value = approx_fun(year, value, rule = 2)) %>%
+      ungroup %>%
+      mutate(fuel = as.character(fuel)) ->
+      L132.GWh_province_F_elec_out
+
+    # Oil and gas does not exist in the CPSY so we will use the CESY for those
+    # Note these are in terms of energy in.  Since we are assuming constant efficiencies for all provinces
+    # mixing these with the elec_out will be fine for approtioning to provinces
+    L101.inNBS_Mtce_province_S_F %>%
+      filter(sector == "electricity" & fuel %in% c("refined liquids", "gas")) ->
+      L132.Mtce_province_F_elec_in
+
+    L132.Mtce_province_F_elec_in[is.na(L132.Mtce_province_F_elec_in)] <- 0
+    L132.Mtce_province_F_elec_in %>%
+      subset(select = -sector) %>%
+      bind_rows(L132.GWh_province_F_elec_out) ->
+      L132.Mix_province_F_elec
+
+    L132.Mix_province_F_elec %>%
+      # In Tibet, refined liquids inherits oil and gas inherits natural gas
+      mutate(fuel = replace(fuel, fuel == "oil" & province == "XZ", "refined liquids"),
+             fuel = replace(fuel, fuel == "natural.gas" & province == "XZ", "gas"),
+             # In Tibet our oil and gas numbers are in electricity out and they need to be energy in in Mtce. This is a total hack but
+             # just divide by the efficiency. Gas is 0.55 and oil is about 0.35
+             tag = 0,
+             tag = replace(tag, fuel == "refined liquids" & province == "XZ", 1),
+             tag = replace(tag, fuel == "gas" & province == "XZ", 2),
+             value = replace(value, tag != 0, 0),
+             copy_rl = value,
+             copy_g = value,
+             copy_rl = copy_rl * 100 * CONV_GWH_EJ * CONV_EJ_MTCE / -0.35,
+             copy_g = copy_g * 100 * CONV_GWH_EJ * CONV_EJ_MTCE / -0.55,
+             copy_rl = replace(copy_rl, tag != 1 , 0),
+             copy_g = replace(copy_g, tag != 2,0),
+             value = value + copy_rl +copy_g) %>%
+      subset(select = c(-tag, -copy_g, -copy_rl)) ->
+      L132.Mix_province_F_elec
+
+    # Aggregate by fuel compute each province's percentage, by fuel
+    L132.Mix_province_F_elec %>%
+      group_by(fuel, year) %>%
+      summarise(sum(value)) %>%
+      ungroup ->
+      L123.Mix_CHINA_F_elect
+    names(L123.Mix_CHINA_F_elect)[3] <- 'Sum'
+
+    L123.pct_province_elec_F <- L132.Mix_province_F_elec %>%
+      left_join_error_no_match(L123.Mix_CHINA_F_elect, by = c('year', 'fuel')) %>%
+      # All historical solar production was central solar
+      mutate(pct = value / Sum,
```

Comment:
line 109-118 not needed: 
`L132.Mix_province_F_elec %>%`
`group_by(fuel, year) %>%`
`mutate(pct = value / sum(value), `